src/parser/mod.rs

```diff
@@ -7928,6 +7930,22 @@ impl<'a> Parser<'a> {
                 vec![]
             };
 
+            let settings = if dialect_of!(self is ClickHouseDialect|GenericDialect)
+                && self.parse_keyword(Keyword::SETTINGS)
+            {
+                let mut key_values: Vec<Setting> = vec![];
+                self.parse_comma_separated(|p| {
+                    let key = p.parse_identifier(false)?;
+                    p.expect_token(&Token::Eq)?;
+                    let value = p.parse_value()?;
+                    key_values.push(Setting { key, value });
+                    Ok(())
+                })?;
+                Some(key_values)
```

Comment:
```suggestion
                let key_values = self.parse_comma_separated(|p| {
                    let key = p.parse_identifier(false)?;
                    p.expect_token(&Token::Eq)?;
                    let value = p.parse_value()?;
                    Ok(Setting { key, value })
                })?;
                Some(key_values)
```
Ah so I think no need to manually build up the list since parse_comma_separated already returns the vec, something like above example?